### PR TITLE
Parsing and creating header and rss file

### PR DESCRIPTION
**Pull request by krumyantsev-sc**

### CODE REVIEW

File: bin/add-feed.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+podcast_id=$1
```
Comment (achernyakevich-sc) [inline bin/add-feed.sh]: Add shebang (shell indicator) like:

```
#!/bin/bash

```

File: bin/add-feed.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+base_directory="../feeds"
```
Comment (achernyakevich-sc) [inline bin/add-feed.sh]: In bash programming naming conventions for variables is snake-bases uppercase. Refactor it.

File: bin/add-feed.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+cd "$base_directory" || exit
+
+if [ ! -d "$podcast_id" ]; then
+    mkdir "$podcast_id"
+fi
```
Comment (achernyakevich-sc) [inline bin/add-feed.sh:27]: But what if folder already exists?

File: bin/add-feed.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+response=$(curl -s -w '%{http_code}' -o "$temp_file" "https://smotrim.ru/podcast/$podcast_id")
+status=$(echo $response | awk '{print $NF}')
+
+if [ "$status" -ne 200 ]; then
+    echo "Error: server responded with status $status"
+    rm -rf "./$podcast_id"
+    exit 1
+fi
+
+html=$(<"$temp_file")
```
Comment (achernyakevich-sc) [inline bin/add-feed.sh]: Can we use `doc2data` to get JSON object representing a podcast and `jq`  to transform JSON to XML?

**Review by achernyakevich-sc — CHANGES_REQUESTED** [review]
Fix obvious and discuss with me about doc2data concept. :)

File: bin/add-feed.sh (hunk 1 above)
Comment (krumyantsev-sc) [reply bin/add-feed.sh]: done

File: bin/add-feed.sh (hunk 3 above)
Comment (krumyantsev-sc) [reply bin/add-feed.sh:27]: If the folder already exists, the script will simply skip the creation step and proceed with the rest of the operations. The if condition checks for the existence of the directory using `[ ! -d "$PODCAST_ID" ]`.

File: bin/add-feed.sh (hunk 2 above)
Comment (krumyantsev-sc) [reply bin/add-feed.sh]: refactored

File: bin/add-feed.sh (hunk 4 above)
Comment (krumyantsev-sc) [reply bin/add-feed.sh]: Implemented

File: bin/add-feed.sh (hunk 3 above)
Comment (achernyakevich-sc) [reply bin/add-feed.sh:27]: No. It should stop and show warning about feed existence.

**Review by achernyakevich-sc — APPROVED** [review]

**Pull request merged**